### PR TITLE
Added location, type, and privacy to completed-docs

### LINKED ISSUE #1713: Add more lintable locations to completed-docs rule
As per the list by HamletDRC in https://github.com/Microsoft/tslint-microsoft-contrib/issues/162, a more complete list of things that could have rules:

* modules
* exported classes
* all classes
* exported interfaces
* interfaces
* interface members
* exported variables
* functions
* exported functions
* public member variables
* all member variables
* public member methods
* all member methods

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Also adds interfaces, enums, types, and a few other doc-able contructs.

Fixes #1713.
Lays the groundwork for #1921.

### CODE REVIEW

File: src/rules/completedDocsRule.ts [hunk 1]
```diff
@@ -19,100 +19,415 @@ import * as ts from "typescript";
...
+export type RequirementDescriptor = IBlockRequirementDescriptor | IClassRequirementDescriptor;
+
+export interface IRequirementDescriptors {
+    [type: string /* DocType */]: RequirementDescriptor;
+}
+
+export type All = typeof Rule.ALL;
+
+export type DocType = All
+    | typeof Rule.ARGUMENT_CLASSES
```
Comment (nchen63) [inline src/rules/completedDocsRule.ts]: isn't `typeof Rule.ARGUMENT_CLASSES` "string"?

Comment (JoshuaKGoldberg) [inline src/rules/completedDocsRule.ts]: What version of TypeScript are you on? For me (VS Code, TypeScript 2.1.X) `typeof Rule.ARGUMENT_CLASSES` correctly resolves to the `"classes"` literal.

Historically, before support for discriminated literal types, I liked having this explicit typing - both to inform the user and for future-proofing the type declarations.

Comment (nchen63) [inline src/rules/completedDocsRule.ts]: I'm using VS Code, TS 2.1.4. So you're saying that `const a: Visibility = "s";` causes a compilation error for you?

Comment (nchen63) [inline src/rules/completedDocsRule.ts]: Did you meant to use `const` instead of `public static`? In that case, I see the string literal type.

Comment (JoshuaKGoldberg) [inline src/rules/completedDocsRule.ts]: Oh, you're totally right. Thanks for the catch! I've switched them to `const`.

**Comment by nchen63** [comment]
I think you need to rebase

**Comment by JoshuaKGoldberg** [comment]
Rebase should be good to go.

**Comment by nchen63** [comment]
I think this may be considered a breaking change if someone already specifies "all" because they'll see new errors. Thoughts?

**Comment by JoshuaKGoldberg** [comment]
Agreed. Changed it to use the current default, so that the new locations are opt-in. For the next TSLint release we could consider expanding default coverage.

**Comment by FDiskas** [comment]
:running_shirt_with_sash: when it will be released?

**Comment by adidahiya** [comment]
@FDiskas it's available since v4.5. you can find out by yourself by clicking through to the squashed commit:

image

and look at the tags list:

image

**Comment by Viktor-Bredihin** [comment]
what's the status? I still can't get it work

**Comment by JoshuaKGoldberg** [comment]
@viktor-bredihin what is your tslint.json config, and what are you trying to lint? Can you post a minimum reproduction we can take a look at? What problems are you seeing?

**Comment by Viktor-Bredihin** [comment]
@JoshuaKGoldberg 
I'm trying to make this rule work '"completed-docs": true', it's not showing any errors in webstorm, however works fine when I use tslint in the console

**Comment by JoshuaKGoldberg** [comment]
Ah - that's not an issue with this rule. That's an issue with the Webstorm TSLint extension.

`completed-docs` is a "type checked" rule ([link to docs](https://palantir.github.io/tslint/usage/type-checking)). That means it uses more expensive APIs from TypeScript. Many extensions don't support this kind of rule for performance reasons.

**Pull request merged**